### PR TITLE
Add optional explicit trust for execution role

### PR DESCRIPTION
**Pull request by michael-ullrich-1010**

### CODE REVIEW

File: modules/execution-role/main.tf [hunk 1]
```diff
@@ -51,6 +51,21 @@ data "aws_iam_policy_document" "lambda" {
     actions = [
       "sts:AssumeRole"
     ]
+    dynamic "condition" {
+      for_each = var.iam_execution_role_explicit_trust == true ? [true] : []
+      content {
+        test     = "ArnEquals"
+        variable = "aws:SourceArn"
```
Comment (JonasNuvibit) [inline modules/execution-role/main.tf:58]: cool wenn das geht 👌

**Review by JonasNuvibit — DISMISSED** [review]

File: modules/execution-role/variables.tf [hunk 2]
```diff
@@ -46,6 +46,12 @@ variable "iam_execution_role_permissions_boundary_arn" {
   }
 }
 
+variable "iam_execution_role_explicit_trust" {
+  description = "Specifies, if the IAM Execution Role shall trust the only the Lambda provisioned by this module."
```
Comment (stefano-franco) [inline modules/execution-role/variables.tf:50]: "Specifies whether the IAM execution role should trust only the Lambda provisioned by this module."

File: variables.tf [hunk 3]
```diff
@@ -310,6 +310,12 @@ variable "iam_execution_role_permissions_boundary_arn" {
   }
 }
 
+variable "iam_execution_role_explicit_trust" {
+  description = "Specifies, if the IAM Execution Role shall trust the only the Lambda provisioned by this module."
```
Comment (stefano-franco) [inline variables.tf:312]: "Specifies whether the IAM execution role should trust only the Lambda provisioned by this module."

File: variables.tf [hunk 4]
```diff
@@ -82,21 +82,19 @@ variable "timeout" {
   }
 }
 
+# https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html
 variable "runtime" {
   description = "Identifier of the function's runtime. See Runtimes for valid values."
   type        = string
 
   validation {
     condition = contains([
```
Comment (stefano-franco) [inline variables.tf:91]: remove this condition to avoid having to manually update the list of runtimes.

**Review by stefano-franco — CHANGES_REQUESTED** [review]

**Pull request closed without merging**